TruePath.Tests/Utils.cs

```diff
@@ -0,0 +1,18 @@
+namespace TruePath.Tests;
+
+public static class Utils
+{
+    internal static IEnumerable<char> MakeNonCanonicalPath(this string path)
+    {
+        foreach (var @char in path)
+        {
+            if (char.IsLetter(@char) && Random.Shared.NextSingle() >= 0.5)
```

Comment:
I would prefer this to switch the case every other character, with no `Random`. And check that `path` is not equal to the resulting string in the end.